File: ruby/texttest_fixture.rb
#!/usr/bin/ruby -w

require File.join(File.dirname(__FILE__), 'gilded_rose')

puts "OMGHAI!"
items = [
  Item.new("+5 Dexterity Vest", 10, 20),
  Item.new("Aged Brie", 2, 0),
  Item.new("Elixir of the Mongoose", 5, 7),
  Item.new("Sulfuras, Hand of Ragnaros", 0, 80),
  Item.new("Sulfuras, Hand of Ragnaros", -1, 80),
  Item.new("Backstage passes to a TAFKAL80ETC concert", 15, 20),
  Item.new("Backstage passes to a TAFKAL80ETC concert", 10, 49),
  Item.new("Backstage passes to a TAFKAL80ETC concert", 5, 49),
  # This Conjured item does not work properly yet
  Item.new("Conjured Mana Cake", 3, 6), # <-- :O
]

days = 2
if ARGV.size > 0
  days = ARGV[0].to_i + 1
end

gilded_rose = GildedRose.new items
(0...days).each do |day|
  puts "-------- day #{day} --------"
  puts "name, sellIn, quality"
  items.each do |item|
    puts item
  end
  puts ""
  gilded_rose.update_quality
end

File: elixir/lib/texttest_fixture.ex
defmodule GildedRose.TextTestFixture do

  def run(days) do
    IO.puts("OMGHAI!")

    items = [
      %Item{name: "+5 Dexterity Vest", sell_in: 10, quality: 20},
      %Item{name: "Aged Brie", sell_in: 2, quality: 0},
      %Item{name: "Elixir of the Mongoose", sell_in: 5, quality: 7},
      %Item{name: "Sulfuras, Hand of Ragnaros", sell_in: 0, quality: 80},
      %Item{name: "Sulfuras, Hand of Ragnaros", sell_in: -1, quality: 80},
      %Item{name: "Backstage passes to a TAFKAL80ETC concert", sell_in: 15, quality: 20},
      %Item{name: "Backstage passes to a TAFKAL80ETC concert", sell_in: 10, quality: 49},
      %Item{name: "Backstage passes to a TAFKAL80ETC concert", sell_in: 5, quality: 49},
      # This Conjured item does not work properly yet
      %Item{name: "Conjured Mana Cake", sell_in: 3, quality: 6}, # <-- :O
    ]

    Enum.reduce(0..days, items, fn day, items ->
      IO.puts "-------- day #{day} --------"
      IO.puts "name, sellIn, quality"
      Enum.each(items, fn item -> IO.puts("#{item.name}, #{item.sell_in}, #{item.quality}") end)
      IO.puts("")
      GildedRose.update_quality(items)
    end)
  end

  def run() do
    run(2)
  end

end
